restart_plugin/mtcp_split_process.c

```diff
@@ -69,23 +78,27 @@ splitProcess(RestoreInfo *rinfo)
   DPRINTF("Initializing Lower-Half Proxy");
   // In startProxy, we fork a child process and exec to lh_proxy.
   // The child process enters libproxy.c:first_constructor() of lh_proxy.
-  // We write rinfo.pluginInfo.memRange to stdin of child process, which
+  // We write memRange to stdin of child process, which
   //   gets copied into lh_info.memRange, inside lh_proxy (see lower-half dir.).
-  // We then read from stdout of child process to populate rinfo.pluginInfo
-  //   with all fields from lh_info.
+  // We then read from stdout of child process to populate lh_info with
+  //   all fields from lh_info in the lower half.
   pid_t childpid = startProxy(rinfo);
   int ret = -1;
   if (childpid > 0) {
     // Parent has read from pipefd_out; child is ready.
     // We then "copy the bits" of the child process into our (parent's) space.
+    // And it is then safe to copy the lh_info from the child proxy to uh.
     // We then kill the child process.
     ret = read_lh_proxy_bits(rinfo, childpid, rinfo->argv[0]);
+    // Populate lh_info now that we have copied the bits.
+    mtcp_memcpy(&lh_info, lh_info_addr, sizeof(lh_info));
```

Comment:
@gc00 
Though this design works fine and there are multiple ways to do a single task, why do we need to populate the whole `lh_info` structure? One can simply use the `lh_info` structure directly from the `lh_proxy's` data section. The `lh_info's` address in the lower half is valid and accessible directly (without needing to switch context) throughout the life of the process.